lms/services/organization.py

```diff
@@ -43,29 +43,38 @@ def get_by_linked_guid(self, guid) -> List[Organization]:
     def get_by_public_id(self, public_id: str) -> Optional[List]:
         """Get an organization by its public_id."""
 
-        return (
-            self._db_session.query(Organization)
-            .filter(Organization.public_id_eq(public_id, self._region))
-            .one_or_none()
-        )
+        return self._search_query(public_id=public_id).one_or_none()
 
-    def search(self, name, limit=100):
+    def search(self, public_id=None, name=None, limit=100) -> List[Organization]:
         """
-        Search organizations.
+        Search for organizations.
+
+        The results are returned as an OR of the specified filters.
 
-        :param name: Match organization by name. Case insensitive.
+        :param public_id: Match on public id
+        :param name: Match organization by name. Case-insensitive.
         :param limit: Limit the number of results
         """
-        return (
-            self._db_session.query(Organization)
-            .filter(
+        return self._search_query(public_id=public_id, name=name).limit(limit).all()
+
+    def _search_query(self, public_id=None, name=None):
```

Comment:
I was going to propose to include ID here but I've seen you done it in a follow up PR.

We could rename this to `organization_query` or something like that? I guess we are "searching" for them in all cases so not a strong suggestion on the name change.

Reply:
My thinking with the name was:

 * You get the noun (organization) from context
 * The verb is "searching" - I know "query" is a pseudomyn of search, but in an SQL context a "query" could write instead of read
 
How about a best of both with `_organization_search_query`?